### ISSUE TITLE
C statements give opaque error message when no semicolon

### ISSUE THREAD

**Issue by SamuelMarks** [semgrep/semgrep#4176]
**Describe the bug**
Semgrep Core ERROR - Pattern parse error: In rule -: Invalid pattern for C

```json
{"code":2,"level":"error","message":"Semgrep Core ERROR - Pattern parse error: In rule your_pattern_id: Invalid pattern for C","rule_id":"your_pattern_id","spans":[{"config_end":{"col":32,"line":3,"offset":60},"config_path":["rules","0","pattern"],"config_start":{"col":12,"line":3,"offset":40}}],"type":"Pattern parse error"}
```

**To Reproduce**
Steps to reproduce the behavior, ideally a link to https://semgrep.dev:

Pattern:
```
int n = strlen(...)
```

C code:
```c
#include <string.h>

int main(void) {
    size_t i;
    for (i=0; i<strlen("OOO"); i++) {}
    int n = strlen(5);
}
```

**Expected behavior**
A clear message suggesting that a semicolon is needed

**Screenshots**
N/A

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [x] P2: regular bug that should get fixed

**Environment**
`brew install` of v0.70.0

**Comment by emjin** [comment]
You need the semicolon so it's a valid statement: https://semgrep.dev/s/pYp0. Hopefully as the C parser matures, we can handle things like this more gracefully (feel free to convert this issue into a "better error message" one).

**Comment by SamuelMarks** [comment]
Thanks that final semicolon did the trick. Unexpected only because the regular `semgrep --lang c -e 'strlen(...)' "$PWD"` didn't require a statement end. But yeah, better error message would be handy for n00bs

**Comment by emjin** [comment]
Absolutely---it's because `strlen(...)` is an expression, but `int $X = $Y` can only be a statement, and the parser expects a semicolon at the end of a statement. Glad it worked!

**Comment by aryx** [comment]
Would be nice also to not have this JSON blob on the upper right and instead highlight the place in the rule that is
wrong.

**Comment by emjin** [comment]
Yes this is something I want to fix in the near future

**Comment by aryx** [comment]
@ievans when you reopen, please assign a priority. @mjambon changed the stale bot to not bother us on issues with a priority (unless it was set to high and nobody did anything on the issue for 2 weeks).

**Comment by ievans** [comment]
I added `planned-project` which also prevents stale bot, but it's good to have the priority too :)
